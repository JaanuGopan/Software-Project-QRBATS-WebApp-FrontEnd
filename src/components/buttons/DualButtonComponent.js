import React, { useState } from "react";

const DualButtonComponent = ({ onSelect, buttonText1, buttonText2 }) => {
  const [selectedButton, setSelectedButton] = useState(1);

  const handleSelect = (button) => {
    setSelectedButton(button);
    onSelect(button);
  };

  return (
    <div className="row align-items-center">
      <div className="col align-self-center">
        <label>{`Select one`}</label>
      </div>
      <div className="col">
        <button
          className={`btn  ${
            selectedButton === 1 ? "btn-primary" : "btn-outline-primary"
          }`}
          onClick={() => handleSelect(1)}
        >
          {buttonText1}
        </button>
      </div>
      <div className="col">
        <button
          className={`btn ${
            selectedButton === 2 ? "btn-primary" : "btn-outline-primary"
          }`}
          onClick={() => handleSelect(2)}
        >
          {buttonText2}
        </button>
      </div>
    </div>
  );
};

export default DualButtonComponent;
