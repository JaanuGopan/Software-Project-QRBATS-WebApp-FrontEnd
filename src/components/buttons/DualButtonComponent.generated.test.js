import renderer from 'react-test-renderer';
import React, { useState } from "react";
import DualButtonComponent from "./DualButtonComponent";

const renderTree = tree => renderer.create(tree);
describe('<DualButtonComponent>', () => {
  it('should render component', () => {
    expect(renderTree(<DualButtonComponent 
    />).toJSON()).toMatchSnapshot();
  });
  
});