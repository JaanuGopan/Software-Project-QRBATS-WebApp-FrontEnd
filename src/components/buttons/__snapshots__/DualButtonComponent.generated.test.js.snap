// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DualButtonComponent> should render component 1`] = `
<div
  className="row m-2   "
>
  <div
    className="col"
  >
    <h6>
      Select one
    </h6>
  </div>
  <div
    className="col"
  >
    <button
      className="btn  btn-primary"
      onClick={[Function]}
    />
  </div>
  <div
    className="col"
  >
    <button
      className="btn btn-outline-primary"
      onClick={[Function]}
    />
  </div>
</div>
`;
